drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/widget/FactPatternWidget.java

```diff
@@ -664,145 +660,26 @@ public void execute() {
 
     private Widget operatorDropDown(final SingleFieldConstraint constraint,
                                     final FlexTable inner,
-                                    final int row,
-                                    final int col) {
-        final HorizontalPanel hp = new HorizontalPanel();
+                                    final int rowIndex,
+                                    final int colIndex) {
+        final HorizontalPanel dropdownContainer = new HorizontalPanel();
         if (!this.readOnly) {
 
-            getOperatorDropDown(constraint,
-                                inner,
-                                row,
-                                col,
-                                new Callback<CEPOperatorsDropdown>() {
-                                    @Override
-                                    public void callback(CEPOperatorsDropdown result) {
-                                        hp.add(result);
-                                    }
-                                });
+            final SingleFieldConstraintOperatorSelectorBuilder operatorSelectorBuilder =
+                    GWT.create(SingleFieldConstraintOperatorSelectorBuilder.class);
+            operatorSelectorBuilder.hasConstraint(constraint);
```

Comment:
I think the builder pattern is great for when you're constructing a complex object with optional **configurations** however in this case they all seem **mandatory** and IDK what value having separate "has" setters has - in fact it make construction error prone as for example if I don't call ```hasPlaceholderForOperatorsDropdown(..)``` then I will get a runtime NPE error. Please either consider replacing all "has" setters with a single "configure(..)" method that takes all mandatory parameters. Or, at least in the ```build()``` method validate all mandatory parameters have been set.